### ISSUE TITLE
odata groupBy causes subquery uses ungrouped column "__select__.author_id" from outer query

### ISSUE THREAD

**Issue by NidhiM9** [NidhiM9/botproj#12]
<html>
<body>
<h3 dir="auto" style="box-sizing: border-box; margin-top: 0px !important; margin-bottom: var(--base-size-16); font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Description of erroneous behaviour</h3><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: var(--base-size-16); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">CDS server crashes with error<span> </span><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">TypeError: transformedQuery.SELECT.from.ref is not iterable</code><span> </span>when trying to<span> </span><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">groupBy</code><span> </span>on a child association.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: var(--base-size-16); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Stacktrace:</p><div class="snippet-clipboard-content notranslate position-relative overflow-auto" style="box-sizing: border-box; position: relative !important; overflow: auto !important; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><pre class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; margin-top: 0px; margin-bottom: var(--base-size-16); overflow-wrap: normal; padding: var(--base-size-16); overflow: auto; line-height: 1.45; color: var(--fgColor-default, var(--color-fg-default)); background-color: var(--bgColor-muted, var(--color-canvas-subtle)); border-radius: 6px;"><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0px; margin: 0px; white-space: pre; background: transparent; border-radius: 6px; word-break: normal; border: 0px; display: inline; overflow: visible; line-height: inherit; overflow-wrap: normal;">[cds] - TypeError: transformedQuery.SELECT.from.ref is not iterable
    at expandColumn (node_modules\@cap-js\db-service\lib\cqn4sql.js:790:41)
    at handleExpand (node_modules\@cap-js\db-service\lib\cqn4sql.js:420:40)
    at node_modules\@cap-js\db-service\lib\cqn4sql.js:354:32
    at getTransformedColumns (node_modules\@cap-js\db-service\lib\cqn4sql.js:382:21)
    at transformSelectQuery (node_modules\@cap-js\db-service\lib\cqn4sql.js:156:41)
    at cqn4sql (node_modules\@cap-js\db-service\lib\cqn4sql.js:73:26)
    at PostgresService.cqn4sql (node_modules\@cap-js\db-service\lib\SQLService.js:375:12)
    at PostgresService.cqn2sql (node_modules\@cap-js\db-service\lib\SQLService.js:351:18)
    at PostgresService.onSELECT (node_modules\@cap-js\db-service\lib\SQLService.js:127:39)
    at PostgresService.onSELECT (node_modules\@cap-js\postgres\lib\PostgresService.js:314:18) {
  id: '1043785',
  level: 'ERROR',
  timestamp: 1719308262527
}
</code></pre><div class="zeroclipboard-container position-absolute right-0 top-0" style="box-sizing: border-box; position: absolute !important; top: 0px !important; right: 0px !important;"><clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0" data-copy-feedback="Copied!" data-tooltip-direction="w" value="[cds] - TypeError: transformedQuery.SELECT.from.ref is not iterable
    at expandColumn (node_modules\@cap-js\db-service\lib\cqn4sql.js:790:41)
    at handleExpand (node_modules\@cap-js\db-service\lib\cqn4sql.js:420:40)
    at node_modules\@cap-js\db-service\lib\cqn4sql.js:354:32
    at getTransformedColumns (node_modules\@cap-js\db-service\lib\cqn4sql.js:382:21)
    at transformSelectQuery (node_modules\@cap-js\db-service\lib\cqn4sql.js:156:41)
    at cqn4sql (node_modules\@cap-js\db-service\lib\cqn4sql.js:73:26)
    at PostgresService.cqn4sql (node_modules\@cap-js\db-service\lib\SQLService.js:375:12)
    at PostgresService.cqn2sql (node_modules\@cap-js\db-service\lib\SQLService.js:351:18)
    at PostgresService.onSELECT (node_modules\@cap-js\db-service\lib\SQLService.js:127:39)
    at PostgresService.onSELECT (node_modules\@cap-js\postgres\lib\PostgresService.js:314:18) {
  id: '1043785',
  level: 'ERROR',
  timestamp: 1719308262527
}" tabindex="0" role="button" style="box-sizing: border-box; position: relative; display: inline-block; padding: 0px !important; font-size: 14px; font-weight: var(--base-text-weight-medium, 500); line-height: 20px; white-space: nowrap; vertical-align: middle; cursor: pointer; user-select: none; border-width: 1px; border-style: solid; border-color: var(--button-default-borderColor-rest, var(--color-btn-border)); border-image: initial; border-radius: 6px; appearance: none; color: var(--button-default-fgColor-rest, var(--color-btn-text)); background-color: var(--button-default-bgColor-rest, var(--color-btn-bg)); box-shadow: var(--button-default-shadow-resting, var(--color-btn-shadow)),var(--button-default-shadow-inset, var(--color-btn-inset-shadow)); transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1), background-color, box-shadow, border-color; margin: var(--base-size-8, 8px) !important;"><svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-copy js-clipboard-copy-icon m-2"><path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z"></path><path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z"></path></svg></clipboard-copy></div></div><h3 dir="auto" style="box-sizing: border-box; margin-top: var(--base-size-24); margin-bottom: var(--base-size-16); font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Detailed steps to reproduce</h3><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: var(--base-size-16); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Try to run an odata query similar to the following:</p><div class="snippet-clipboard-content notranslate position-relative overflow-auto" style="box-sizing: border-box; position: relative !important; overflow: auto !important; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><pre class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; margin-top: 0px; margin-bottom: var(--base-size-16); overflow-wrap: normal; padding: var(--base-size-16); overflow: auto; line-height: 1.45; color: var(--fgColor-default, var(--color-fg-default)); background-color: var(--bgColor-muted, var(--color-canvas-subtle)); border-radius: 6px;"><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0px; margin: 0px; white-space: pre; background: transparent; border-radius: 6px; word-break: normal; border: 0px; display: inline; overflow: visible; line-height: inherit; overflow-wrap: normal;">$apply: filter(foo eq 'bar')/groupby((user/name),aggregate(amount with sum as totalAmount))
</code></pre><div class="zeroclipboard-container position-absolute right-0 top-0" style="box-sizing: border-box; position: absolute !important; top: 0px !important; right: 0px !important;"><clipboard-copy aria-label="Copy" class="ClipboardButton btn js-clipboard-copy m-2 p-0" data-copy-feedback="Copied!" data-tooltip-direction="w" value="$apply: filter(foo eq 'bar')/groupby((user/name),aggregate(amount with sum as totalAmount))" tabindex="0" role="button" style="box-sizing: border-box; position: relative; display: inline-block; padding: 0px !important; font-size: 14px; font-weight: var(--base-text-weight-medium, 500); line-height: 20px; white-space: nowrap; vertical-align: middle; cursor: pointer; user-select: none; border-width: 1px; border-style: solid; border-color: var(--button-default-borderColor-rest, var(--color-btn-border)); border-image: initial; border-radius: 6px; appearance: none; color: var(--button-default-fgColor-rest, var(--color-btn-text)); background-color: var(--button-default-bgColor-rest, var(--color-btn-bg)); box-shadow: var(--button-default-shadow-resting, var(--color-btn-shadow)),var(--button-default-shadow-inset, var(--color-btn-inset-shadow)); transition: color 80ms cubic-bezier(0.33, 1, 0.68, 1), background-color, box-shadow, border-color; margin: var(--base-size-8, 8px) !important;"><svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true" class="octicon octicon-copy js-clipboard-copy-icon m-2"><path d="M0 6.75C0 5.784.784 5 1.75 5h1.5a.75.75 0 0 1 0 1.5h-1.5a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-1.5a.75.75 0 0 1 1.5 0v1.5A1.75 1.75 0 0 1 9.25 16h-7.5A1.75 1.75 0 0 1 0 14.25Z"></path><path d="M5 1.75C5 .784 5.784 0 6.75 0h7.5C15.216 0 16 .784 16 1.75v7.5A1.75 1.75 0 0 1 14.25 11h-7.5A1.75 1.75 0 0 1 5 9.25Zm1.75-.25a.25.25 0 0 0-.25.25v7.5c0 .138.112.25.25.25h7.5a.25.25 0 0 0 .25-.25v-7.5a.25.25 0 0 0-.25-.25Z"></path></svg></clipboard-copy></div></div><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: var(--base-size-16); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">If you change<span> </span><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">groupby((user/name)</code><span> </span>with<span> </span><code class="notranslate" style="box-sizing: border-box; font-family: var(--fontStack-monospace, ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace); font-size: 11.9px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">groupby((user_ID)</code><span> </span>it works as expected.</p><h3 dir="auto" style="box-sizing: border-box; margin-top: var(--base-size-24); margin-bottom: var(--base-size-16); font-size: 1.25em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Details about your project</h3><markdown-accessiblity-table data-catalyst="" style="box-sizing: border-box; display: block; margin-bottom: 0px !important; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">
Package | Version
-- | --
OData version | v4
@cap-js/cds-typer | 0.22.0
@cap-js/cds-types | 0.2.0
@cap-js/postgres | 1.8.0
@sap/cds | 7.9.2
@sap/cds-compiler | 4.9.4
@sap/cds-dk | 7.9.3
@sap/cds-dk (global) | 7.9.2
@sap/cds-fiori | 1.2.4
@sap/cds-foss | 5.0.1
@sap/cds-mtxs | 1.18.1
@sap/eslint-plugin-cds | 3.0.4
Node.js | v20.13.1

</markdown-accessiblity-table>
</body>
</html>

**Comment by NidhiM9** [comment]
Hi, could you please let me know which version of the db-service you are using (its not part of the version table you've provided)

**Comment by NidhiM9** [comment]
My db-service is at v1.10.1.
BTW, just checked capire and this query should work correctly: https://cap.cloud.sap/docs/advanced/odata#example-1

**Comment by NidhiM9** [comment]
Also, is odata v4 filter in implemented? Doesn't seem to work for me either.
My CDS config:

    "features": {
        "assert_integrity": "db",
        "odata_new_parser": true,
        "odata_new_adapter": false,
        "string_decimals": false
    }

**Comment by NidhiM9** [comment]
My db-service is at v1.10.1. BTW, just checked capire and this query should work correctly: https://cap.cloud.sap/docs/advanced/odata#example-1

yes, this should be supported. I have identified your issue and prepared a fix in https://github.com/cap-js/cds-dbs/pull/709
Once this change is merged, I will ship a release.

Thank you for reporting this issue 😊

Also, is odata v4 filter in implemented? Doesn't seem to work for me either. My CDS config:

    "features": {
        "assert_integrity": "db",
        "odata_new_parser": true,
        "odata_new_adapter": false,
        "string_decimals": false
    }
Sorry, I can't help you as I'm no expert with the ODATA adapter. Maybe @NidhiM9  can help you out.

**Comment by NidhiM9** [comment]
unfortunately this is still not working:

Query:

$apply: filter(start ge 2024-06-01T00:00:00Z and start le 2024-06-30T23:59:59Z)/groupby((user/name),aggregate(amount with sum as totalAmount))
Error:

[cds] - error: subquery uses ungrouped column "__select__.user_id" from outer query
    at node_modules\pg\lib\client.js:526:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.all (node_modules\@cap-js\postgres\lib\PostgresService.js:174:26)
    at async PostgresService.onSELECT (node_modules\@cap-js\db-service\lib\SQLService.js:132:16)
    at async next (node_modules\@sap\cds\lib\srv\srv-dispatch.js:68:17)
    at async PostgresService.handle (node_modules\@sap\cds\lib\srv\srv-dispatch.js:66:10)
    at async ApplicationService.<anonymous> (node_modules\@sap\cds\libx\_runtime\common\generic\crud.js:73:16)
    at async next (node_modules\@sap\cds\lib\srv\srv-dispatch.js:68:17)
    at async ApplicationService.handle (node_modules\@sap\cds\lib\srv\srv-dispatch.js:66:10)
    at async ApplicationService.handle (node_modules\@sap\cds\libx\_runtime\common\Service.js:87:16) {
  length: 152,
  severity: 'ERROR',
  code: '42803',
  detail: undefined,
  hint: undefined,
  position: '255',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'parse_agg.c',
  line: '1450',
  routine: 'check_ungrouped_columns_walker',
  query: 'SELECT to_jsonb(root.*) as _json_ FROM (SELECT jsonb(root."user") as "user",root.totalAmount as "totalAmount" FROM (SELECT (SELECT to_jsonb(user2.*) as _json_ FROM (SELECT user2.name as "name" FROM (SELECT user2.name FROM CrmService_Users as user2 WHERE __select__.user_ID =
 user2.ID LIMIT $1) as user2) as user2) as "user",sum(__select__.amount) as totalAmount FROM (SELECT TimeTransactions.ID,TimeTransactions.oldCrmId,TimeTransactions.createdAt,TimeTransactions.createdBy,TimeTransactions.modifiedAt,TimeTransactions.modifiedBy,TimeTransactions.timeVault_
ID,TimeTransactions.amount,TimeTransactions.user_ID,TimeTransactions.comment_ID,TimeTransactions.start,TimeTransactions."end",TimeTransactions.type,TimeTransactions.topic_ID FROM CrmService_TimeTransactions as TimeTransactions WHERE TimeTransactions.start >= $2 and TimeTransactions.start <= $3 ORDER BY TimeTransactions.ID ASC) as __select__ left JOIN CrmService_Users as "user" ON "user".ID = __select__.user_ID GROUP BY "user".name LIMIT $4) as root) as root\n' +
    ' ^',
  id: '1366272',
  level: 'ERROR',
  timestamp: 1719383922927
}

**Comment by NidhiM9** [comment]
I was able to reproduce your issue. The problem is - as the error states - that the group by clause of the outer query is missing the author.ID which is used to build the correlated subquery for the expand.

Long story short, I was able to reproduce your issue, in my bookshop.

http://localhost:4004/admin/Books?$apply=filter(title%20ne%20%27bar%27)/groupby((author/name))
observe the error: subquery uses ungrouped column "__select__.author_id" from outer query
subquery:

SELECT author2.name
FROM AdminService_Authors as author2
WHERE __select__.author_ID = author2.ID
LIMIT $1
main query __select__:

SELECT (
-- expand subquery from above
) as author
FROM (
   …
) as __select__
left JOIN …
GROUP BY author.name -- this is missing the author.ID
For me, including the author/ID in my odata request, did the trick

- http://localhost:4004/admin/Books?$apply=filter(title%20ne%20%27bar%27)/groupby((author/name))
+ http://localhost:4004/admin/Books?$apply=filter(title%20ne%20%27bar%27)/groupby((author/name,author/ID))
{
  "value": [
    {
      "author": {
        "ID": 150,
        "name": "Edgar Allen Poe"
      }
    },
    {
      "author": {
        "ID": 107,
        "name": "Charlotte Brontë"
      }
    },
    {
      "author": {
        "ID": 101,
        "name": "Emily Brontë"
      }
    },
    {
      "author": {
        "ID": 170,
        "name": "Richard Carpenter"
      }
    }
  ],
  "@odata.context": "$metadata#Books(author(name,ID))"
}
Could you try that? I will ask my colleagues if we can give you some convenience for this in the future, as this is a technical detail which one might not expect.

**Comment by NidhiM9** [comment]
While yes, including author/ID, is a valid workaround which works for me as well, groupby(author/name) should also work since it is documented in the [capire docs](https://cap.cloud.sap/docs/advanced/odata#example-1) and [odata docs](https://devblogs.microsoft.com/odata/aggregation-extensions-in-odata-asp-net-core/#groupby-transformation).
In this particular case groupby(author/name) and groupby(author/name, author/ID) are equivalent but there are cases when you want to group by attributes which will not have the equivalent aggregation result.

**Comment by NidhiM9** [comment]
we have shipped a fix for your issue. Let me know if it works out for you :)

**Comment by NidhiM9** [comment]
Sorry, for late response. After upgrading to db-service v1.12.0 everything seems to be working as it should.
Thank you for the fix 🙂
